EIPS/eip-5553.md

```diff
@@ -0,0 +1,260 @@
+        songIds.increment();
+
+        SongRegistration newSong = new SongRegistration(
+                                        songIds.current(),
+                                        address(this),
+                                        _params,
+                                        address(comp),
+                                        address(rec)
+                                    );
+    }
+}
+
+
+```
+
+## Rationale
+
+### A more robust royalty interest structure
+
+Current royalty implementations deal only with a single type of royalty payments: NFT sales. They also only allow a single type of royalty - i.e you cannot have different royalty interest owners receive royalty based on different licensing types. 
+In other words, currently, a royalty split works the same way no matter what type of purchase or license deal has happened, for all parties involved.
+
+With this proposal, we allow multiple **types** of royalty interests. A classic case is the music industry in which you have writing/composition royalties, and recording/master royalties. Different licensing types will pay different percentages to different parties based on context.
+
+In the case of a song cover, a license payment formula can be created so that that 
+a) Original work's writers get paid for using the lyrics or composition of the song
+b) recording artists of the original song do not get paid since their recording is not used
+c) recording artists of the new work will get paid
+d) there are no writing royalties for the creators of the cover.
+
+Moreover, we have single structure that connects to all types of royalty types, and allows finding them easily.
+
+### A more robust future for licensing
+By separating the canonical version of the work from its various licensed uses (NFT purchase, streaming, usage of art etc..) we are allowing a new ecosystem of various license types and payment types for a single work, based on the same standard.
+
+### A single pointer to the works metadata
+The work points to metadata housed in IPFS or arweave and allows changing it and keeping track of the changes in a simple and standard way. Today the only metadata standard is NFT metadata extension, but we do not know which standard the document adheres to. With different work types we can forumlate different metadata standards for different work types and have a simple easy place to do discover attached metadata.
+
+### Allowing people from creative industries to put their work on chain
+
+We foresee new marketplaces that allow registering your work, all base don the IWorksRegistration standard, but implemented for different types of works: Books, video, music and more. These marketplaces will support familiar royalty interest use cases, with the added benefits of transparent, fast and permissionless web3 infrastructure.
+
+## Security Considerations
+
+There are no security considerations related directly to the implementation of this standard.
```

Comment:
[peer review] I'd personally feel there shall be security considerations for distributing royalties, such as preventing reentry etc. But it's up to you to decide.